spark-cassandra-connector-java/src/main/java/com/datastax/spark/connector/japi/RDDAndDStreamCommonJavaFunctions.java

```diff
@@ -177,7 +177,7 @@ public WriterBuilder withBatchSize(BatchSize batchSize) {
                 return withWriteConf(
                     new WriteConf(batchSize, writeConf.batchGroupingBufferSize(), writeConf.batchGroupingKey(),
                         writeConf.consistencyLevel(), writeConf.parallelismLevel(), writeConf.throughputMiBPS(),
-                        writeConf.ttl(), writeConf.timestamp()));
+                        writeConf.ttl(), writeConf.timestamp(), writeConf.taskMetricsEnabled()));
```

Comment:
Not now, but for the future: I think this whole fragment is too repeatable and written in Scala-style which doesn't play nice with Java. I think we need to implement those `withXYZ` methods in a standard Java way - that is by setting a private variable and then eventually having a build() method that creates a final `WriteConf`.